addons/web/models/base_document_layout.py

```diff
@@ -181,9 +215,66 @@ def _parse_logo_colors(self, logo=None, white_threshold=225):
     @api.model
     def action_open_base_document_layout(self, action_ref=None):
         if not action_ref:
-            action_ref = 'base.action_base_document_layout_configurator'
+            action_ref = 'web.action_base_document_layout_configurator'
         return self.env["ir.actions.actions"]._for_xml_id(action_ref)
 
     def document_layout_save(self):
         # meant to be overridden
         return self.env.context.get('report_action') or {'type': 'ir.actions.act_window_close'}
+
+    def _get_asset_style(self):
+        """
+        Compile the style template. It is a qweb template expecting company ids to generate all the code in one batch.
+        We give a useless company_ids arg, but provide the PREVIEW_ID arg that will prepare the template for
+        '_get_css_for_preview' processing later.
+        :return:
+        """
+        template_style = self.env.ref('web.styles_company_report', raise_if_not_found=False)
+        if not template_style:
+            return b''
+
+        company_styles = template_style._render({
+            'company_ids': [self],
+            'PREVIEW_ID': '__PREVIEW_ID__'
+        })
+
+        return company_styles
+
+    @api.model
+    def _get_css_for_preview(self, scss, new_id):
+        """
+        Compile the scss into css. Since we had to get around company ids, the actual id has been replaced by
+        __PREVIEW_ID__. It get swapped by new_id, the company id used in the preview... Which is in reality the wizard
+        id. Which means it is not yet an id in DB, but a newId object.
+        """
+        css_code = self._compile_scss(scss)
+        css_code = css_code.replace('__PREVIEW_ID__', str(new_id))
```

Comment:
I don't like this because it forces very specific code in the template (a hack....)
I suppose in the template, company.id resolves to an object (a NewID in this case), and you want it to resolve to the function__str__ return ?
Isn't it possible to use already existing newID feature to discriminate, in the template, between
- id is really empty
- id is a NewID
- id is a integer

?